### ISSUE TITLE
Moving to Elpa

### ISSUE THREAD

**Issue by Ergus** [tumashu/ivy-posframe#95]
Hi:

In the mailing list we have been talking these days about some modernization changes in Emacs. Today ivy-posframe called out attention as it seems an interesting interface alternative to the ones available either in vanilla emacs and elpa.

Could you consider to move ivy-posframe to the official elpa to become one of the official packages? We know the process could be a bit cumbersome; but it really worth the effort and we are available to help as much as possible.

Pending from your reply,
Best regards,
Ergus

**Comment by tumashu** [comment]
wow, it may need some work, for i merge many pr, i do not know them gpl paper sign status.

**Comment by tumashu** [comment]
The friends in below list, do you have asigned gpl paper?

@abo-abo           (ok)
@conao3            (  )
@tttuuu888         (  )
@CeleritasCelery   (  )
@xeijin            (  )
@gagbo             (  )
@noctuid           (  )
@ramsayleung       (  )
@drrlvn            (  )

**Comment by conao3** [comment]
Good proposal, I have already signed gpl paper.

**Comment by tttuuu888** [comment]
@tumashu Yes I have. Thank you.

**Comment by gagbo** [comment]
I didn't. How should I proceed ?

**Comment by tumashu** [comment]
@gagbo https://www.gnu.org/software/emacs/manual/html_node/emacs/Copyright-Assignment.html#Copyright-Assignment

**Comment by tumashu** [comment]
Copyright assignment is a simple process. Residents of some countries can do it entirely electronically. We can help you get started, including sending you the forms you should fill, and answer any questions you may have (or point you to the people with the answers), at the emacs-devel@gnu.org mailing list.

**Comment by tumashu** [comment]
> Copyright assignment is a simple process. Residents of some countries can do it entirely electronically. We can help you get started, including sending you the forms you should fill, and answer any questions you may have (or point you to the people with the answers), at the [emacs-devel@gnu.org](mailto:emacs-devel@gnu.org) mailing list.

**Comment by Ergus** [comment]
> I didn't. How should I proceed ?

Hi:

The process takes some daysto complete, but is actually simple. 

https://www.gnu.org/software/emacs/manual/html_node/emacs/Copyright-Assignment.html

Just send an email in your name to emacs-devel@gnu.org asking for a Copyright Assignment for an Elpa package. You will receive a small form asking basic data: Name, package you will contribute to, email and not much more. And then you'll receive a document you should print-sign-scan-emailback.

And that's it.

**Comment by CeleritasCelery** [comment]
I have signed the paper work.

**Comment by Ergus** [comment]
Hi, no progress on this? :(

**Comment by gagbo** [comment]
I have signed a week ago, still waiting on news

**Comment by tumashu** [comment]
@Ergus What is the standard of "small contributions" ?  code line number?

**Comment by tumashu** [comment]
@Ergus 

1. @ramsayleung  has contributed  >15 lines code,  they are  defcustom variables, small contributions?
2. @noctuid ( ) has contributed  > 15 lines code,  many of them come from  ivy.el and swiper.el.   so I do not know how to deal with it.

**Comment by ramsayleung** [comment]
> The friends in below list, do you have asigned gpl paper?

Hi @tumashu, I don't have asigned gpl paper for now, I am not familiar with Elpa, it's necessary to have one to move ivy-posframe to the official elpa? I just committed 24 lines code to `ivy-posframe`, is there something to do with me?

**Comment by tumashu** [comment]
@ramsayleung  I am waiting  @Ergus 's suggestion :-)

**Comment by Ergus** [comment]
> @ramsayleung I am waiting @Ergus 's suggestion :-)

Hi: I replied on yesterday but somehow the message it has disappeared.

@ramsayleung The paperwork is required in any case :( but it is indeed very simple (just a couple of emails). If you don't want to do the copyright there is also a possibility to do a disclaim (which is even simpler to do); but IMO the copyright it better because enables yo continue contributing any time in the future if the package goes to elpa.

Whatever you decide it starts sending an email  as explained in the links above.

Best,
Ergus

**Comment by ramsayleung** [comment]
> @ramsayleung The paperwork is required in any case :( but it is indeed very simple (just a couple of emails). If you don't want to do the copyright there is also a possibility to do a disclaim (which is even simpler to do); but IMO the copyright it better because enables yo continue contributing any time in the future if the package goes to elpa.

@Ergus Thanks for your head's up, I know the necessity of a paperwork now. Is there any guide about how to sign a GPL paper? I have never signed a GPL paperwork before.

**Comment by Ergus** [comment]
https://www.gnu.org/software/emacs/manual/html_node/emacs/Copyright-Assignment.html#Copyright-Assignment

Just write to the emacs mailing list and as for a copyright assignment for an elpa package, they will replly with the small form to fill.

**Comment by tumashu** [comment]
@Ergus If we can not contact with @noctuid, how we should do?  elimit his commit?

**Comment by Ergus** [comment]
> @Ergus If we can not contact with @noctuid, how we should do? elimit his commit?

Hi:

You must ask in the mailing list how to proceed; maybe they'll provide an alternative. Please email yourself to the mailing list as the project head: emacs-devel@gnu.org and explain the situation.

Send also the link to the project and all the basic information that could be important for them to give an accurate reply.

See you there,
Ergus

**Comment by tumashu** [comment]
@abo-abo Is it possible to merge ivy-posframe to ivy？

**Comment by noctuid** [comment]
I will send an email asking about assignment. Feel free to ping me in a few days if I haven't commented here again.

**Comment by ramsayleung** [comment]
I have signed the paper work :)

**Comment by tumashu** [comment]
cool!

**Comment by tumashu** [comment]
@noctuid @gagbo  Hi friends,  gpl paper status?  :-)

**Comment by noctuid** [comment]
I will have to find someone at my company who can sign a disclaimer that the company won't try to claim my contributions. I am looking into it.

**Comment by gagbo** [comment]
I received the paper earlier today, I'm done with the paperwork

**Comment by tumashu** [comment]
@noctuid OK

**Comment by abo-abo** [comment]
> @abo-abo Is it possible to merge ivy-posframe to ivy？

It's possible. But why would you like to do it? I think it's better to have a separate repo and list of issues.

**Comment by Ergus** [comment]
Pong :-)

Hi @tumashu 

You already have some repos in Elpa. So you know how to proceed. Do you need something else why did you ping in the mailing list?? @noctuid is already done with the paperwork?

**Comment by tumashu** [comment]
@Ergus  Because  the below line in README : 

```
There is no approval process for GNU Elpa packages.  Still,
you must send an email to emacs-devel for several reasons:

- Notifying other developers;
- Making sure the package doesn't break FSF rules;
- Checking if the package is not reinventing the wheel;
- Ensuring that first-time developers are doing it right.
```

As for @noctuid's contribution,  I think ok now, for the below three reasons:
1. https://lists.gnu.org/archive/html/emacs-devel/2020-09/msg01982.html
2. https://lists.gnu.org/archive/html/emacs-devel/2020-09/msg02049.html
3. I have rewrite most of his code according to newest ivy.el ivy-avy.el and swiper.el,  should < 15 lines now.
